src/BookDao.java

```diff
@@ -0,0 +1,20 @@
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+
+public class BookDao {
+public static int save(String callno,String name,String author,String publisher,int quantity){
+	int status=0;
+	try{
+		Connection con=DB.getConnection();
+		PreparedStatement ps=con.prepareStatement("insert into books(callno,name,author,publisher,quantity) values(?,?,?,?,?)");

```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer

**Problem**
This line of code contains a resource that might not be closed properly. Resource leaks can cause your system to slow down or crash.

**Fix**
Consider closing the following resource: *ps*. The resource is referenced in statements at the following lines: 9, 10, 11, 12, 13, 14, 15. The resource closure statement is at line: 16. There are other execution paths that do not contain closure statements, e.g., when exception is thrown by PreparedStatement.setInt. Either a) close *ps* in a try-finally block or b) close the resource by declaring *ps* in a try-with-resources block.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).